### DISCUSSION TITLE
Some confusion about Proj1_Part3

### DISCUSSION THREAD

**Discussion by Lurkers990909** [sist-cs120/project-wiki#21] [Project 1 Q&A]
When we test our program on one device by a sound card, it performs really well. The receiver can receive the complete input in most cases. 
However, when we try to run "Transimtter" and "Receiver" on 2 differnent devices, the difference between "OUTPUT.TXT" and "INPUT.TXT" is big. What can we do to improve its performance?

**Comment by yhyan96** [comment]
There are two aspects affecting the performance.

1. The frequency of two sound cards are not the same.
    Although you have set them to the same frequency, e.g., 48000 Hz, they are still slightly different, e.g., one is 48000 Hz and another one is 47000 Hz. It will make the decode out of sync.
    To relieve this problem, you should avoid long frame. Also, you need to design a better modulation scheme.
    If your problem is caused by it, the errors may seldom appear at the beginning of each frame.

2. The gain of two sound cards are different.
    Although the speaker plays the sound with the same amplitude, different microphones may record the sound with different amplitudes. 
    You should plot the sound recorded by the new microphone, and the threshold for demodulation should be tuned.
    If your problem is caused by it, the errors may distribute more uniformly.
